gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/AvroKeyRecordReaderWrapper.java

```diff
@@ -0,0 +1,84 @@
+/* (c) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.mapreduce.avro;
+
+import java.io.IOException;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.avro.mapreduce.AvroJob;
+import org.apache.avro.mapreduce.AvroKeyRecordReader;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.CombineFileSplit;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+
+
+/**
+ * A subclass of {@link org.apache.hadoop.mapreduce.RecordReader} that contains an
+ * {@link org.apache.avro.mapreduce.AvroKeyRecordReader} object.
+ *
+ * This class is necessary since {@link org.apache.avro.mapreduce.AvroKeyRecordReader} cannot be directly used
+ * to construct {@link org.apache.hadoop.mapreduce.lib.input.CombineFileRecordReader}, because it doesn't
+ * have a constructor with signature (CombineFileSplit, TaskAttemptContext, Integer).
+ *
+ * @author ziliu
+ */
+public class AvroKeyRecordReaderWrapper extends RecordReader<AvroKey<GenericRecord>, NullWritable> {
```

Comment:
Would this make sense as an extension of AvroKeyRecordReader instead of a wrapper? A lot of methods are just calling the wrapped method.